### PR TITLE
Add a server-sided way to remove color codes from incoming chat messages

### PR DESCRIPTION
**Pull request by red-001**
These code be generated by CSM, a modded client or just copy and pasted by the player.

### CODE REVIEW

File: src/server.cpp [hunk 1]
```diff
@@ -2871,12 +2871,16 @@ void Server::handleChatInterfaceEvent(ChatEvent *evt)
 }
 
 std::wstring Server::handleChat(const std::string &name, const std::wstring &wname,
-	const std::wstring &wmessage, bool check_shout_priv, RemotePlayer *player)
+	const std::wstring &wmessage_input, bool check_shout_priv, RemotePlayer *player)
```
Comment (SmallJoker) [inline src/server.cpp]: Using a reference here is nonsense if we're going to copy the whole string anyway.

Comment (red-001) [inline src/server.cpp]: fixed

**Comment by rubenwardy** [comment]
I'd err towards enabling this by default

**Comment by nerzhul** [comment]
@rubenwardy restriction enable should be admin responsibility, we just offer the feature

**Comment by red-001** [comment]
Considering you can enable this without even needing to restart the server I don't think it's a major issue

**Pull request merged**